Backpack/src/main/java/net/skyscanner/backpack/flare/BpkFlare.kt

```diff
@@ -151,20 +175,31 @@ open class BpkFlare @JvmOverloads constructor(
       pointerPosition.offset
     }
 
-    val yStart = height - pointerMask.height
+    val (yEnd, yStart) = if (pointerDirection == PointerDirection.DOWN) {
```

Comment:
Instead of using pairs you can just clip rect in this conditions. If at any point we'll be needed to draw it on left/right it'll be more scalable.

Reply:
Not sure I see the benefit, if we need different left/right logic we'll see what to do then, although that is very unlikely. This will duplicate the clip logic in both branches when the only thing that really changes is the y start/end.

Reply:
Doesn't apply anymore with the logic to rotate the canvas